### PR TITLE
check for + fix correct shebang in scripts in check_header

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by JensTimmerman** [comment]
the check wasn't really needed because vsc-install will always write it out when installing.

**Comment by boegel** [comment]
Well, it's good to catch these issues before installing, no?

Also, explain me how https://github.ugent.be/hpcugent/vsc-filesystems-quota/pull/40 happened then?

File: lib/vsc/install/headers.py [hunk 1]
```diff
@@ -180,12 +180,39 @@ def check_header(filename, script=False, write=False):
...
+                     file_ext, filename)
+
+        changed = shebang != orig_shebang
+
+    if orig_shebang is not None:
         # original position
-        header_end_pos += 1 + len(shebang)  # 1 is from splitted newline
+        header_end_pos += 1 + len(orig_shebang)  # 1 is from splitted newline
 
         if 'python' in shebang and shebang != SHEBANG_ENV_PYTHON:
```
Comment (boegel) [inline lib/vsc/install/headers.py]: @JensTimmerman It only changed the shebang if i) if there was a shebang (see also `shebang is not None` above), ii) if it has `python` in it and it wasn't the expected `#!/usr/bin/env python`

Comment (JensTimmerman) [reply lib/vsc/install/headers.py]: that's only the code that does the checking and auto fixing of headers.

I'm saying that vsc-install adds the shebang sat install time, not at test time

https://github.com/hpcugent/vsc-install/blob/be3a2b0261cd497b4dd9e286242f94becd46cc83/lib/vsc/install/shared_setup.py#L621

Comment (JensTimmerman) [reply lib/vsc/install/headers.py]: but currently only for python scripts, not for bash scripts. might need to be extended for that.

Comment (JensTimmerman) [reply lib/vsc/install/headers.py]: I seem to be confused, that line also only does the replacement of env to non env
there used to be code around somewhere that added shebangs, but I can't seem to find it

Comment (boegel) [reply lib/vsc/install/headers.py]: If it was there, https://github.ugent.be/hpcugent/vsc-filesystems-quota/pull/40 wouldn't have happened.

Also, there was no test that checked whether Python scripts with missing shebang was being corrected, now there is.

Comment (itkovian) [reply lib/vsc/install/headers.py]: The python scripts had no shebang line after installation (and did not have any in the repo) when https://github.ugent.be/hpcugent/vsc-filesystems-quota/pull/39 was merged.

**Comment by itkovian** [comment]
+1.

**Pull request merged**